File: 2211-k-radius-subarray-averages/k-radius-subarray-averages.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        res = [-1] * n
        window_size = 2 * k + 1
        window_sum = 0

        if window_size > n:
            return res
        
        prefix = [nums[0]]
        for i in range(n):
            prefix.append(prefix[-1] + nums[i])
        
        for i in range(k, n - k):
            curr_sum = prefix[i + k + 1] - prefix[i - k]
            res[i] = curr_sum // window_size
        return res
# [100000 100000]

File: 2211-k-radius-subarray-averages/k-radius-subarray-averages.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        int window_size = 2 * k + 1;
        long window_sum = 0;
        int n = nums.size();
        vector<int> res(n, -1);

        if (n < window_size) {
            return res;
        }

        for (int i = 0; i < n; ++i) {
            window_sum += nums[i];

            if (i - window_size >= 0) {
                window_sum -= nums[i - window_size];
            }

            if (i >= window_size - 1) {
                res[i - k] = window_sum / window_size;
            }
        }

        return res;
    }
};